uint8 start_state
uint8 goal_state
